Adds Powerfist Augment Weapon

Ryan180602 opened the pull request:
:cl: Ryan180602, AzlanOnPc
rscadd: Add Pneumatic Powerfist
/:cl:

So, my first big* additive PR. With Azlan's wonderful sprites.
(*its not that big)

A bit spaghetti code, still needs a bit ironing out.
Everything else works, though. I'm willing to change things out for balance/un-spaghettifying reasons. And well, this was a good learning experience anyways.

Different pressure values change the amount of damage you do (and of course the amount of gas in air-tank consumed).
At  ultimate pressure value, it can yeet people when hit, and can open airlocks the same way an exosuit clamp can.

The powerfist only takes the small emergency tanks.

Highest force the powerfist can have is 47
Lowest is 5

Two yeets/door breaks before a new tank needs to be installed.

juxjux9930 commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+	item_state = "powerfist"
+	name = "pneumatic powerfist"
+	icon = 'icons/obj/augment.dmi'
+	desc = "A strong, hydraulic operated powerfist. Packs quite the punch with other utility uses."
```
if it uses an air tank isn't it pneumatic

Ryan180602 replied:
forgot to change that out, thankies

juxjux9930 commented on code/modules/research/designs/designs_mechfab.dm:
```diff
@@ -532,6 +532,13 @@
 	id = "augment_wristblade"
 
+/datum/design/item/mechfab/augment/armblade/popout_shotgun
+	name = "Hydraulic Powerfist"
```
same think. air ppwer go pneu

Ryan180602 replied:
also didnt change the datum path, thanku

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+
+
+/obj/item/powerfist/attackby(obj/item/W as obj, mob/user as mob) //load tank
+	if(!tank && istype(W,/obj/item/tank/emergency) && user.unEquip(W, src))
```
In the event it already has a tank, I'd prefer a user feedback message appear along the lines of `SPAN_WARNING("\The [src] already has \a [tank] installed.")`

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+/obj/item/powerfist/attackby(obj/item/W as obj, mob/user as mob) //load tank
+	if(!tank && istype(W,/obj/item/tank/emergency) && user.unEquip(W, src))
+		tank = W
+		user.visible_message("[user] jams [W] into [src]'s valve and twists it closed.","You jam [W] into [src]'s valve and twist it closed.")
```
- Multi-lined `visible_message()` calls are easier to read due to their length.
- Use `SPAN_*()` define wrapped for displayed messages (NOTE: For `examine()` text this isn't necessary unless you want to highlight something as good or bad).
- Use `\the` and `\a` prefixes where applicable.

```suggestion
		user.visible_message(
			SPAN_NOTICE("\The [user] jams \a [W] into \the [src]'s valve and twists it closed."),
			SPAN_NOTICE("You jam \the [W] into \the [src]'s valve and twist it closed.")
		)
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+		return
+
+
+/obj/item/powerfist/verb/set_pressure() //set tank pressure
```
- Verb's name is self-explanatory.

```suggestion
/obj/item/powerfist/verb/set_pressure()
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+	var/N = input("Percentage of tank used per hit:","[src]") as null|anything in possible_pressure_amounts
+	if (N)
+		pressure_setting = N
+		to_chat(usr, "You dial the pressure valve to [pressure_setting]%.")
```
```suggestion
		to_chat(usr, SPAN_NOTICE("You dial \the [src]'s pressure valve to [pressure_setting]%."))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+
+/obj/item/powerfist/attack_hand(mob/living/user) //remove tank
+	if(!tank)
+		to_chat(user, "There's no tank in [src].")
```
```suggestion
		to_chat(user, SPAN_WARNING("There's no tank in \the [src]."))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+	if(!tank)
+		to_chat(user, "There's no tank in [src].")
+		return
+	to_chat(user, "You twist the valve and pop the tank out of [src].")
```
- If inserting a tank creates a visible message, removing it should be equivalent.

```suggestion
	user.visible_message(
		SPAN_NOTICE("\The [user] twists the valve and pops \a [tank] out of \the [src]."),
		SPAN_NOTICE("You twist the valve and pop \the [tank] out of \the [src].")
	)
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+	if(tank)
+		to_chat(user, "The tank dial reads [tank.air_contents.return_pressure()] kPa.")
+	else
+		to_chat(user, "Nothing is attached to the tank valve!")
```
May be worth adding a line indicating what kind of tank is installed.

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+				var/obj/machinery/door/airlock/A = target
+				if (!A.operating && !A.locked)
+					if (A.welded)
+						A.visible_message(SPAN_DANGER("\The [user] forces their fingers in through the welded metal, beginning to pry \the [A] open!"))
```
- The message should indicate the powerfist is being used for this

```suggestion
						A.visible_message(
							SPAN_DANGER("\The [user] forces the fingers of \the [src] in through the welded metal, beginning to pry \the [A] open!")
						)
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+				if (!A.operating && !A.locked)
+					if (A.welded)
+						A.visible_message(SPAN_DANGER("\The [user] forces their fingers in through the welded metal, beginning to pry \the [A] open!"))
+						if (do_after(user, 13 SECONDS, A) && !A.locked)
```
- There's a public visible message, so the progress bar should also be public.
- Lack of unique acts on interactions causes shenanigans. May be worth using `DO_BOTH_UNIQUE_ACT` instead, but unsure how that may affect other door `do_after()` interactions.

```suggestion
						if (do_after(user, 13 SECONDS, A, DO_DEFAULT | DO_PUBLIC_PROGRESS | DO_USER_UNIQUE_ACT) && !A.locked)
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+	augment_flags = AUGMENT_MECHANICAL | AUGMENT_SCANNABLE
+
+
+/obj/item/powerfist/attackby(obj/item/W as obj, mob/user as mob) //load tank
```
- `as xyz` isn't necessary with type-casted parameters. (I believe this is a thing from older beyond versions)

```suggestion
/obj/item/powerfist/attackby(obj/item/W, mob/user) //load tank
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+	tank.remove_air(lost_gas)
+
+
+/obj/item/powerfist/afterattack(var/atom/target, var/mob/living/user, var/inrange, var/params)
```
- `var/` isn't necessary in parameters

```suggestion
/obj/item/powerfist/afterattack(atom/target, mob/living/user, inrange, params)
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+
+
+/obj/item/powerfist/afterattack(var/atom/target, var/mob/living/user, var/inrange, var/params)
+	if (tank && !istype(target, /turf))
```
- Use `isturf()` for turf checks.

```suggestion
	if (tank && !isturf(target))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+							A.update_icon()
+							playsound(A, 'sound/effects/meteorimpact.ogg', 100, 1)
+							playsound(A, 'sound/machines/airlock_creaking.ogg', 100, 1)
+							A.visible_message(SPAN_DANGER("\The [user] tears \the [A] open!"))
```
```suggestion
							A.visible_message(SPAN_DANGER("\The [user] tears \the [A] open with \a [src]!"))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+							A.set_broken(TRUE)
+							return
+					else
+						A.visible_message(SPAN_DANGER("\The [user] pries their fingers in, beginning to force \the [A]!"))
```
```suggestion
						A.visible_message(SPAN_DANGER("\The [user] pries the fingers of \a [src] in, beginning to force \the [A]!"))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+							return
+					else
+						A.visible_message(SPAN_DANGER("\The [user] pries their fingers in, beginning to force \the [A]!"))
+						if ((A.is_broken(NOPOWER) || do_after(user, 10 SECONDS, A)) && !(A.operating || A.welded || A.locked))
```
```suggestion
						if ((A.is_broken(NOPOWER) || do_after(user, 10 SECONDS, A, DO_DEFAULT | DO_PUBLIC_PROGRESS | DO_USER_UNIQUE_ACT)) && !(A.operating || A.welded || A.locked))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+								addtimer(CALLBACK(A, /obj/machinery/door/airlock/.proc/open, TRUE), 0)
+								if(!A.is_broken(NOPOWER))
+									A.set_broken(TRUE)
+								A.visible_message(SPAN_DANGER("\The [user] forces \the [A] open!"))
```
```suggestion
								A.visible_message(SPAN_DANGER("\The [user] forces \the [A] open with \a [src]!"))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+								addtimer(CALLBACK(A, /obj/machinery/door/airlock/.proc/close, TRUE), 0)
+								if (!A.is_broken(NOPOWER))
+									A.set_broken(TRUE)
+								A.visible_message(SPAN_DANGER("\The [user] forces \the [A] closed!"))
```
```suggestion
								A.visible_message(SPAN_DANGER("\The [user] forces \the [A] closed with \a [src]!"))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+									A.set_broken(TRUE)
+								A.visible_message(SPAN_DANGER("\The [user] forces \the [A] closed!"))
+				if (A.locked)
+					to_chat(user, SPAN_NOTICE("The airlock's bolts prevent it from being forced."))
```
- Failure feedback messages should use warning.

```suggestion
					to_chat(user, SPAN_WARNING("The airlock's bolts prevent it from being forced."))
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+			if (istype(target, /mob/living) && pressure_setting > 20)
+				var/mob/living/A = target
+				A.throw_at(get_edge_target_turf(user, user.dir), pressure_setting/10, pressure_setting/10) //penultimate/ultimate settings yeets people
+				user.visible_message(SPAN_DANGER("\The [user] batters [A], sending them flying!"))
```
```suggestion
				user.visible_message(
					SPAN_DANGER("\The [user] batters \the [A] with \a [src], sending them flying!"),
					SPAN_WARNING("You batter \the [A] with \the [src], sending them flying!")
				)
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,143 @@
+				user.visible_message(SPAN_DANGER("\The [user] batters [A], sending them flying!"))
+		if (tank.air_contents.return_pressure() < 200)
+			playsound(user, 'sound/machines/ekg_alert.ogg', 50)
+			user.visible_message(SPAN_DANGER("The dial flashes red, indicating an empty tank!"))
```
```suggestion
			user.visible_message(SPAN_DANGER("\The [src]'s dial flashes red, indicating an empty tank!"))
```

Ryan180602 commented:
made all the suggested changes plus a minor tweak to how the gas loss is handled in the after_attack

SierraKomodo reviewed (changes requested):
Both `/obj/item/organ/internal/augment/active/item/powerfist` and `/obj/item/powerfist` are missing sprites - You have the icon and item states defined, but the .dmi file hasn't actually had the sprites added to it.

The lack of sprite also makes the deploy button unclickable so I'll have to retest that once the sprites are fixed.

SierraKomodo reviewed (changes requested):
- When there's no tank attached, clicking someone on harm intent performs an attack animation but displays no chat messages and has no actual effect. The effects with no tank should be identical to the effects with an empty tank, but possibly with a different message indicating the tank is missing instead of empty.
- Clicking the deployed powerfist with an invalid tank presents no user feedback message. This should at least display a `SPAN_WARNING()` saying the tank is too big/doesn't fit if it's not the right tank type.
- The fist appears to deal the same amount of damage with a tank installed whether the tank is empty or not.

Ryan180602 commented:
1. If a tank is not installed, the weapon's force is 8. If a tank is installed and kept at 10% pressure value, it jumps to 9.5. When the tank beeps and says its empty, the force turns back down to 8. Verified through VV. It's probably the fact that the 1.5 change in force doesn't seem like much. I can try lower the no-tank/pressureless tank damage?

2. Put a feedback message for bigger tanks. 

3. Made it so that without a tank, the powerfist does a base damage of 8. Would it still need a user feedback for no tank? Or is this fine?

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+/obj/item/powerfist/attackby(obj/item/W, mob/user) //load tank
+	if (tank)
+		user.visible_message(SPAN_WARNING("\The [src] already has \a [tank] installed."))
+	else if (!tank && istype(W,/obj/item/tank/emergency) && user.unEquip(W, src))
+		tank = W
+		user.visible_message(
+			SPAN_NOTICE("\The [user] jams \a [W] into \the [src]'s valve and twists it closed."),
+			SPAN_NOTICE("You jam \the [W] into \the [src]'s valve and twist it closed.")
+		)
+		playsound(user, 'sound/effects/refill.ogg', 50, 1, -6)
+		update_icon()
+		update_force()
+	else if (!istype(W,/obj/item/tank/emergency))
+		user.visible_message(
+			SPAN_WARNING("\The [W] is too big for \the [src]'s valve!")
+		)
```
Checking `!tank` after the `if (tank)` check is redundant. Suggestion also places the `istype()` check before the actual processing to further reduce the number of redundant conditions needed.

```suggestion
	else if (!istype(W,/obj/item/tank/emergency))
		user.visible_message(
			SPAN_WARNING("\The [W] is too big for \the [src]'s valve!")
		)
	else if (user.unEquip(W, src))
		tank = W
		user.visible_message(
			SPAN_NOTICE("\The [user] jams \a [W] into \the [src]'s valve and twists it closed."),
			SPAN_NOTICE("You jam \the [W] into \the [src]'s valve and twist it closed.")
		)
		playsound(user, 'sound/effects/refill.ogg', 50, 1, -6)
		update_icon()
		update_force()
```

SierraKomodo commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+					SPAN_DANGER("\The [user] batters \the [A] with \a [src], sending them flying!"),
+					SPAN_WARNING("You batter \the [A] with \the [src], sending them flying!")
+				)
+		return ..()
+	else
+		return ..()
```
```suggestion
	return ..()
```

SierraKomodo reviewed (dismissed):
Also see my ping in `#code`

Ryan180602 commented:
As discussed, added a 3 second time needed to insert/take out tanks.

Didn't implement the adding tanks while fist is on the ground, it seemed a bit too complex for me.

Added a codex entry for a tiny bit of lore + how to use the thing. Willing to cut the lore out if it doesn't seem to be fitting.

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+	desc = "A strong, pneumatic powerfist. Packs quite the punch with other utility uses."
+	base_parry_chance = 12
+	force = 5
+	attack_cooldown = 1.5*DEFAULT_WEAPON_COOLDOWN
```
```suggestion
	attack_cooldown = 1.5 * DEFAULT_WEAPON_COOLDOWN
```

Spookerton replied:
style since we're here

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+	hitsound = 'sound/effects/bamf.ogg'
+	attack_verb = list("smashed", "bludgeoned", "hammered", "battered")
+
+	var/punch_pressure
+	var/adjusted_force
+	var/obj/item/tank/tank = null
+	var/pressure_setting = 10
+	var/possible_pressure_amounts = list(10,20,30,50)
```
```suggestion
	var/obj/item/tank/tank
	var/pressure_setting
	var/list/possible_pressure_amounts = list(10, 20, 30, 50)
```

Spookerton replied:
remove intermediate values

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+	var/obj/item/tank/tank = null
+	var/pressure_setting = 10
+	var/possible_pressure_amounts = list(10,20,30,50)
+
```
```suggestion

/obj/item/powerfist/Initialize()
	. = ..()
	if (ispath(tank))
		tank = new tank (src)
	if (!pressure_setting)
		pressure_setting = possible_pressure_amounts[1]
	update_force()
```

Spookerton replied:
allows powerfist items to configure subtypes

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+	augment_flags = AUGMENT_MECHANICAL | AUGMENT_SCANNABLE
+
+
+/obj/item/powerfist/attackby(obj/item/W, mob/user) //load tank
+	if (tank)
+		user.visible_message(SPAN_WARNING("\The [src] already has \a [tank] installed."))
+	else if (!istype(W,/obj/item/tank/emergency))
+		user.visible_message(
+			SPAN_WARNING("\The [W] is too big for \the [src]'s valve!")
+		)
+	else if (do_after(user, 3 SECONDS, W, DO_DEFAULT | DO_USER_UNIQUE_ACT | DO_USER_SAME_HAND))
+		user.unEquip(W, src)
+		tank = W
+		user.visible_message(
+			SPAN_NOTICE("\The [user] jams \a [W] into \the [src]'s valve and twists it closed."),
+			SPAN_NOTICE("You jam \the [W] into \the [src]'s valve and twist it closed.")
+		)
+		playsound(user, 'sound/effects/refill.ogg', 50, 1, -6)
+		update_icon()
+		update_force()
```
```suggestion
 /obj/item/powerfist/attackby(obj/item/I, mob/user)
	if (!istype(I, /obj/item/tank/emergency))
		if (istype(I, /obj/item/tank))
			to_chat(user, SPAN_WARNING("\The [I] is too big. Find a smaller tank."))
		return
	if (tank)
		to_chat(user, SPAN_WARNING("\The [src] already has \a [tank] installed."))
		return
	user.visible_message(
		SPAN_ITALIC("\The [user] starts connecting \a [I] to \his [src]."),
		SPAN_ITALIC("You start connecting \the [I] to \the [src]."),
		range = 5
	)
	if (!do_after(user, 3 SECONDS, I, DO_DEFAULT | DO_USER_UNIQUE_ACT | DO_USER_SAME_HAND))
		return
	if (!user.unEquip(I, src))
		return
	user.visible_message(
		SPAN_ITALIC("\The [user] finishes connecting \a [I] to \his [src]."),
		SPAN_NOTICE("You finish connecting \the [I] to \the [src]."),
		range = 5
	)
	playsound(user, 'sound/effects/refill.ogg', 50, 1, -6)
	tank = I
	update_force()
	update_icon()
```

Spookerton replied:
reorder conditions, depth
visible_message -> to_chat for things only the user should see

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+		update_force()
+
+
+/obj/item/powerfist/proc/update_force()
+	if (!tank)
+		force = 5
+	else
+		punch_pressure = tank.air_contents.return_pressure()*pressure_setting/100
+		adjusted_force = (punch_pressure*tank.volume)/425
+		if (tank.air_contents.return_pressure() > 210)
+			force = adjusted_force
+		else
+			force = 5
+		return
```
```suggestion
/obj/item/powerfist/proc/update_force()
	var/pressure = tank?.air_contents?.return_pressure()
	if (pressure > 210)
		force = (pressure * pressure_setting * 0.01) * (tank.volume / 425)
	else
		force = 5
```

Spookerton replied:
remove intermediate stored values, null coerces to 0

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+		return
+
+
+/obj/item/powerfist/verb/set_pressure()
+	set name = "Set Valve Pressure"
+	set category = "Object"
+	set src in range(0)
+	var/N = input("Percentage of tank used per hit:","[src]") as null|anything in possible_pressure_amounts
+	if (N)
+		pressure_setting = N
+		to_chat(usr, SPAN_NOTICE("You dial \the [src]'s pressure valve to [pressure_setting]%."))
+		update_force()
+		return
```
```suggestion
/obj/item/powerfist/proc/set_pressure()
	set name = "Set Powerfist Pressure"
	set desc = "Set the powerfist's tank output pressure."
	set category = "Object"
	set src in range(0)
	var/N = input("Percentage of tank used per hit:", "[src]") as null | anything in possible_pressure_amounts
	if (isnull(N))
		return
	pressure_setting = N
	to_chat(usr, SPAN_NOTICE("You dial \the [src]'s pressure valve to [pressure_setting]%."))
	update_force()


/obj/item/powerfist/attack_self(mob/living/carbon/human/user)
	set_pressure()
```

Spookerton replied:
set_pressure becomes a verbified proc; use-in-hand on powerfist calls it.
null gate instead of falsy gate; could conceivably have a 0 option later

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+		return
+
+
+/obj/item/powerfist/attack_hand(mob/living/user) //remove tank
+	if(!tank)
+		to_chat(user, SPAN_WARNING("There's no tank in \the [src]."))
+		return
+	if(do_after(user, 3 SECONDS, src, DO_DEFAULT | DO_SHOW_PROGRESS | DO_USER_UNIQUE_ACT | DO_USER_SAME_HAND))
+		user.visible_message(
+			SPAN_NOTICE("\The [user] twists the valve and pops \a [tank] out of \the [src]."),
+			SPAN_NOTICE("You twist the valve and pop \the [tank] out of \the [src].")
+		)
+		user.put_in_hands(tank)
+		tank = null
+		playsound(loc, 'sound/effects/spray3.ogg', 50)
+		update_icon()
+		update_force()
```
```suggestion
/obj/item/powerfist/attack_hand(mob/living/user)
	if (!tank)
		to_chat(user, SPAN_WARNING("There's no tank in \the [src]."))
		return
	user.visible_message(
		SPAN_ITALIC("\The [user] starts disconnecting \a [tank] from \his [src]."),
		SPAN_ITALIC("You start disconnecting \the [tank] from \the [src]."),
		range = 5
	)
	if (!do_after(user, 3 SECONDS, src, DO_DEFAULT | DO_SHOW_PROGRESS | DO_USER_UNIQUE_ACT | DO_USER_SAME_HAND))
		return
	user.visible_message(
		SPAN_ITALIC("\The [user] finishes disconnecting \a [tank] from \his [src]."),
		SPAN_NOTICE("You finish disconnecting \the [tank] from \the [src]."),
		range = 5
	)
	user.put_in_hands(tank)
	playsound(loc, 'sound/effects/spray3.ogg', 50)
	tank = null
	update_icon()
	update_force()
```

Spookerton replied:
re-order conditions & depth

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+/obj/item/powerfist/on_update_icon()
+	..()
+	if (tank)
+		overlays+=image(icon, "powerfist_tank")
```
```suggestion
		overlays += image(icon, "powerfist_tank")
```

Spookerton replied:
style since we're here

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+	if (tank)
+		overlays+=image(icon, "powerfist_tank")
+	else
+		overlays-=image(icon,"powerfist_tank")
```
```suggestion
		overlays -= image(icon, "powerfist_tank")
```

Spookerton replied:
style since we're here

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+
+/obj/item/powerfist/examine(mob/living/user, distance)
+	. = ..()
+	if(distance > 2)
```
```suggestion
	if (distance > 2)
```

Spookerton replied:
style since we're here

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+	if(distance > 2)
+		return
+	to_chat(user, "The valve is dialed to [pressure_setting]%.")
+	if(tank)
```
```suggestion
	if (tank)
```

Spookerton replied:
style since we're here

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+
+
+/obj/item/powerfist/proc/no_pressure()
+	if (tank.air_contents.return_pressure() < 210)
+		playsound(usr, 'sound/machines/ekg_alert.ogg', 50)
+		usr.visible_message(SPAN_DANGER("\The [src]'s dial flashes red, indicating an empty tank!"))
+		update_force()
+		return
```
```suggestion
	if (tank && tank.air_contents?.return_pressure() < 210)
		playsound(usr, 'sound/machines/ekg_alert.ogg', 50)
		to_chat(usr, SPAN_WARNING("\The pressure dial on \the [src] flashes a warning: it's out of gas!"))
		update_force()
```

Spookerton replied:
gating in case we use this elsewhere
trailing return

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+
+
+/obj/item/powerfist/proc/gas_loss()
+	var/lost_gas = tank.air_contents.total_moles*(pressure_setting/100)
+	tank.remove_air(lost_gas)
+	return
```
```suggestion
	if (tank?.air_contents)
		var/lost_gas = tank.air_contents.total_moles * pressure_setting * 0.01
		tank.remove_air(lost_gas)
```

Spookerton replied:
tank & mix gate

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+							A.visible_message(SPAN_DANGER("\The [user] forces \the [A] closed with \a [src]!"))
+			if (A.locked)
+				to_chat(user, SPAN_WARNING("The airlock's bolts prevent it from being forced."))
+			return
```
```suggestion
```

Spookerton replied:
extraneous trailing returns

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,178 @@
+					SPAN_DANGER("\The [user] batters \the [A] with \a [src], sending them flying!"),
+					SPAN_WARNING("You batter \the [A] with \the [src], sending them flying!")
+				)
+	return ..()
```
```suggestion
	return ..()


/obj/item/powerfist/prepared
	tank = /obj/item/tank/emergency/oxygen/double

/obj/item/organ/internal/augment/active/item/powerfist/prepared
	item = /obj/item/powerfist/prepared

/obj/item/device/augment_implanter/powerfist
	augment = /obj/item/organ/internal/augment/active/item/powerfist/prepared

```

Spookerton replied:
just throwing these in here for easy testing

Ryan180602 commented:
made necessary changes + the test ones too (I was intending on having it be antag buyable eventually, as well)

Ryan180602 commented:
actually fixed, now.

Spookerton commented on code/modules/augment/active/powerfist.dm:
```diff
@@ -0,0 +1,208 @@
+		force = 5
+
+
+/obj/item/powerfist/proc/set_pressure()
+	set name = "Set Powerfist Pressure"
+	set desc = "Set the powerfist's tank output pressure."
+	set category = "Object"
+	set src in range(0)
```
```suggestion
/obj/item/powerfist/verb/set_pressure_verb()
	set name = "Set Powerfist Pressure"
	set desc = "Set the powerfist's tank output pressure."
	set category = "Object"
	set src in range(0)
	set_pressure()


/obj/item/powerfist/proc/set_pressure()
```

Spookerton replied:
I am a dumb. Should be fine after this.

Ryan180602 replied:
fixed, my bad as well, sorry.

Ryan180602 commented:
Nooot sure why the code check is failing.

SierraKomodo commented on code/modules/codex/entries/augments.dm:
```diff
@@ -38,3 +38,14 @@
+	lore_text = "<p>The AG-23 Pneumatic Powerfist was first devised as a portable door-cracking/obstacle clearing tool to be used by emergency rescue teams. Eventually, it found its way into more public hands through one way or another, and is now mostly used by private security teams, special police breacher squads, wealthy mob enforcers, and is sometimes even seen in the hands of SolGov Defense Fo ...
+	\
+	<p>While it is recommended to use the power-fist at it's lower settings for doors and airlocks, a higher setting is provided in some models for tougher obstacles.</p>"
+	mechanics_text = "The powerfist deals a base 5 points of damage without a tank, or when the tank has an air pressure of below 210 kPa. Activate the powerfist in hand to change the pressure setting; which changes the amount of damage the powerfist does. \
+	At 30% and 50%; the powerfist will be able to tear open welded/unwelded airlocks, barring the secure command airlocks. At 50%, the powerfist is also able to throw people a few tiles away.\
+	\
+	The powerfist only accepts the small emergency air tanks."
```
This will render as one big blob instead of separated paragraphs if you don't include the `<p></p>` tags.
```suggestion
	mechanics_text = "<p>The powerfist deals a base 5 points of damage without a tank, or when the tank has an air pressure of below 210 kPa. Activate the powerfist in hand to change the pressure setting; which changes the amount of damage the powerfist does.</p> \
	<p>At 30% and 50%; the powerfist will be able to tear open welded/unwelded airlocks, barring the secure command airlocks. At 50%, the powerfist is also able to throw people a few tiles away.</p>\
	\
	<p>The powerfist only accepts the small emergency air tanks.</p>"
```

Ryan180602 committed:
add powerfist
